core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2771,6 +2780,84 @@ private CompiledQuery vacuum(SqlExecutionContext executionContext) throws SqlExc
         throw SqlException.$(lexer.lastTokenPosition(), "'partitions' expected");
     }
 
+    private CompiledQuery snapshotDatabase(SqlExecutionContext executionContext) throws SqlException {
+
+        executionContext.getCairoSecurityContext().checkWritePermission();
+        CharSequence snapshotRoot = configuration.getSnapshotRoot();
+        if (null == snapshotRoot) {
+            throw CairoException.instance(0).put("Snapshots are disabled, no snapshot root directory is configured in the server configuration ['cairo.sql.snapshot.root' property]");
+        }
+        CharSequence tok = expectToken(lexer, "'prepare' or 'commit'");
+        if (Chars.equalsLowerCaseAscii(tok, "prepare")) {
+
+            if (lockedReaders.size() > 0) {
+                throw SqlException.position(lexer.lastTokenPosition()).put("Another snapshot command in progress");
+            }
+
+            path.of(configuration.getSnapshotRoot()).slash();
+            configuration.getSnapshotDirTimestampFormat().format(
+                    configuration.getMicrosecondClock().getTicks(),
+                    configuration.getDefaultDateLocale(),
+                    null,
+                    path);
+
+            int snapshotLen = path.length();
+            if (ff.exists(path.slash$())){
```

Comment:
I'd say don't fail, add `.1` etc to the end if you want format to be configurable